### PR TITLE
test: make `createTestDir` a general test utility

### PR DESCRIPTION
**Pull request by G-Rath**
We've already got this duplicated in two places, and I have need for it in an upcoming PR and @robramsaynz should be using it in #797 so let's move this into `testutility`

### CODE REVIEW

**Review by another-rex — APPROVED** [review]

File: internal/testutility/utility.go [hunk 1]
```diff
@@ -57,3 +57,18 @@ func ValueIfOnWindows(win, or string) string {
 
 	return or
 }
+
+// CreateTestDir makes a temporary directory for use in testing that involves
+// writing and reading files from disk - make sure to call the cleanup function!
+func CreateTestDir(t *testing.T) (string, func()) {
+	t.Helper()
```
Comment (spencerschrock) [inline internal/testutility/utility.go:64]: `t.Cleanup` would be helpful here.
https://pkg.go.dev/testing#T.Cleanup

no need to make caller remember to call cleanup function

Comment (G-Rath) [reply internal/testutility/utility.go:64]: oh nice - and yeah I just realised `t.TempDir` is a thing though sadly it doesn't support a prefix which I personally like to easily identify and manually clean up in the case of a failure or debugging.

Oh well, still probably best to just switch to using that

Comment (spencerschrock) [reply internal/testutility/utility.go:64]: Sounds reasonable. At least `t.Cleanup` sounds useful in this case.

**Review by another-rex — APPROVED** [review]
Thanks!

**Pull request merged**